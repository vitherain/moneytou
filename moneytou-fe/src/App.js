import React from 'react';
import './App.scss';

function App() {
  return (
    <div className="app">
      <div id="side-menu">
        Side menu
      </div>
      <div id="screen-content">
        Screen content
      </div>
    </div>
  );
}

export default App;
