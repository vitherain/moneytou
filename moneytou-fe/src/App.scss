.app {
  display: flex;
  min-height: 100%;
  flex-flow: row nowrap;

  &--light {
    background-color: #ffffff;
  }

  &--dark {
    background-color: #1E2125;
    color: rgb(218, 221, 225);
  }

  .side-nav {
    flex: 0 1 30%;
    min-height: 100%;
  }

  .screen-content {
    flex: 0 1 70%;
    min-height: 100%;
  }
}
