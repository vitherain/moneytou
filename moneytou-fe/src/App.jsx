import React from 'react';
import './App.scss';
import SideNav from './core/components/SideNav';
import ScreenContent from './core/components/ScreenContent';

function App() {
  const theme = 'light';

  return (
    <div className={`app app--${theme}`}>
      <SideNav className="side-nav" />
      <ScreenContent className="screen-content" />
    </div>
  );
}

export default App;
