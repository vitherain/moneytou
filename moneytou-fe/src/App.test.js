import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('<App /> spec', () => {
  it('renders the component', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('renders side nav and main body', () => {
    const { container } = render(<App />);
    expect(container.querySelector('.app .side-nav')).toBeInTheDocument();
    expect(container.querySelector('.app .screen-content')).toBeInTheDocument();
  });
});
