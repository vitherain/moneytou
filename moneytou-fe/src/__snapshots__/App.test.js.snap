// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> spec renders the component 1`] = `
<div>
  <div
    class="app app--light"
  >
    <div
      class="side-nav"
    >
      Side nav content
    </div>
    <div
      class="screen-content"
    >
      Screen content
    </div>
  </div>
</div>
`;
